### PR TITLE
Resolve interaction delays for Test Users dropdown

### PR DESCRIPTION
**Pull request by meckila-navapbc**
### Description
Resolve interaction delays for test users dropdown 

### Acceptance Criteria
- [ ] Code compiles correctly
- [ ] Can select and type test user without delay after clicking into dropdown 

### Testing Plan
1. Go to `/test/users`
2. Click into dropdown
3. Verify options load without delay 
3. Begin typing user name
4. Verify that options load without delay

### User Facing Changes
 - [ ] Screenshots of UI changes added to PR & Original Issue

### CODE REVIEW

File: client/app/test/TestUsers.jsx [hunk 1]
```diff
@@ -248,7 +271,7 @@ TestUsers.propTypes = {
   epTypes: PropTypes.array.isRequired,
   userDisplayName: PropTypes.string,
   dropdownUrls: PropTypes.array,
-  userSession: PropTypes.object.isRequired,
+  userSession: PropTypes.object,
```
Comment (meckila-navapbc) [inline client/app/test/TestUsers.jsx:274]: This can be nil

File: client/app/test/TestUsers.jsx [hunk 2]
```diff
@@ -11,151 +11,173 @@ import NavigationBar from '../components/NavigationBar';
...
+        } else {
+          matchByInclusion.push(option);
+        }
+      }
+    }
 
-    const tabs = this.props.appSelectList.map((app) => {
-      let tab = {};
+    return [...matchByStart, ...matchByInclusion];
+  }, [inputValue]);
```
Comment (jcq) [inline client/app/test/TestUsers.jsx:110]: Shouldn't `userOptions` be part of the dependency array here?

File: client/app/test/TestUsers.jsx [hunk 3]
```diff
@@ -11,151 +11,173 @@ import NavigationBar from '../components/NavigationBar';
...
+      setReseedingError(err);
+      setIsReseeding(false);
     });
-  }
+  };
+
+  const userOptions = props.testUsersList.map((user) => ({
+    value: user.id,
+    label: `${user.css_id} at ${user.station_id} - ${user.full_name}`
+  }));
```
Comment (jcq) [inline client/app/test/TestUsers.jsx:86]: This `map` operation should probably be within the `useMemo` for `filteredUserOptions`, as it's only used there; as it is, this is getting recalculated on every render.

File: client/app/test/TestUsers.jsx (hunk 2 above)
Comment (jcq) [reply client/app/test/TestUsers.jsx:110]: (Moot if you move initialization of `userOptions` into this `useMemo`)

**Review by nanotone — APPROVED** [review]
:shipit:

**Pull request merged**